### PR TITLE
Replace EARLY after includes by `#pragma asmproc recurse`

### PR DESCRIPTION
**Pull request by EllipticEllipsis**
I think this is better than the current way, but will probably be replaced by a standalone script with the same functionality.

I also rewrote the if-else chain in the for loop with some `continue`s to a little to make it a bit more readable, at least to me. I can revert that commit if it's preferred to keep the current structure.

Okay for both OoT and MM after tweaking their makefiles.

### CODE REVIEW

File: asm_processor.py [hunk 1]
```diff
@@ -832,44 +833,56 @@ def parse_source(f, opt, framepointer, mips1, input_enc, output_enc, out_depende
...
+            continue
+
+        if line == '#pragma INCLUDE_EARLY':
+            # C includes qualified as
+            # #pragma INCLUDE_EARLY
+            # #include "file.c"
+            # will be processed recursively when encountered
+            is_early_include = True
+            continue
+        elif is_early_include:
```
Comment (simonlindholm) [inline asm_processor.py:863]: error out if the line doesn't match `#include "..."`

File: asm_processor.py [hunk 2]
```diff
@@ -832,44 +833,56 @@ def parse_source(f, opt, framepointer, mips1, input_enc, output_enc, out_depende
...
+            with open(fname, encoding=input_enc) as f:
+                for line2 in f:
+                    global_asm.process_line(line2.rstrip(), output_enc)
+            src, fn = global_asm.finish(state)
+            output_lines[-1] = ''.join(src)
+            asm_functions.append(fn)
+            global_asm = None
+            continue
+
+        if line == '#pragma INCLUDE_EARLY':
```
Comment (simonlindholm) [inline asm_processor.py:856]: `#pragma asmproc recurse`/`follow`/similar to make the syntax clearer? I'm slightly partial to `follow`

File: asm_processor.py [hunk 3]
```diff
@@ -832,44 +833,56 @@ def parse_source(f, opt, framepointer, mips1, input_enc, output_enc, out_depende
...
-                # This is a hack to replace all floating-point numbers in an array of a particular type
-                # (in this case CutsceneData) with their corresponding IEEE-754 hexadecimal representation
-                if cutscene_data_regexpr.search(line) is not None:
-                    is_cutscene_data = True
-                elif line.endswith("};"):
-                    is_cutscene_data = False
-                if is_cutscene_data:
-                    raw_line = re.sub(float_regexpr, repl_float_hex, raw_line)
-                output_lines[-1] = raw_line
+            continue
```
Comment (simonlindholm) [inline asm_processor.py:836]: I think I'd prefer to keep all of these as a big if-elif-chain. Nice to turn `else: if ...` here into elif to reduce indentation level though.

File: asm_processor.py (hunk 1 above)
Comment (EllipticEllipsis) [reply asm_processor.py:857]: Added. I could print the line number too?

**Pull request merged**